src/crypto/qat/qcccrypto.cc

```diff
@@ -0,0 +1,475 @@
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 Intel Corporation
+ *
+ * Author: Ganesh Mahalingam <ganesh.mahalingam@intel.com>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ */
+
+#include "qcccrypto.h"
+#include  <iostream>
+#include "string.h"
+#include <pthread.h>
+#include "common/debug.h"
+
+// -----------------------------------------------------------------------------
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+#undef dout_prefix
+#define dout_prefix _prefix(_dout)
+
+static ostream&
+_prefix(std::ostream* _dout)
+{
+  return *_dout << "QccCrypto: ";
+}
+// -----------------------------------------------------------------------------
+
+/*
+ * Error function when stuff fails during initialization
+*/
+#define ERROUT \
+  qaeMemDestroy(); \
+is_init = false; \
+init_stat = stat; \
+goto init_end;
+
+/*
+ * Poller thread & functions
+*/
+static pthread_mutex_t qccAllocMutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t qcc_eng_mutex = PTHREAD_MUTEX_INITIALIZER;
+unsigned int QccCrypto::cypoll = 0;
+
+void* QccCrypto::qccCyPoll(CpaInstanceHandle instHandle)
+{
+  cypoll = 1;
+  while(cypoll)
+  {
+    icp_sal_CyPollInstance(instHandle, 0);
+    usleep(qccSleepDuration);
+  }
+
+  pthread_exit(NULL);
+}
+
+CpaStatus QccCrypto::qccCyStartPoll(int entry)
+{
+  CpaInstanceInfo2 info;
+  CpaStatus pollstat = CPA_STATUS_SUCCESS;
+
+  pollstat = cpaCyInstanceGetInfo2(qccInst->cyInstHandles[entry], &info);
+  if((pollstat == CPA_STATUS_SUCCESS) && (info.isPolled == CPA_TRUE))
+  {
+    if(pthread_create(&cypollthreads[entry], NULL, *qccCyPoll,
+                       qccInst->cyInstHandles[entry]) != 0)
+    {
+      dout(1) << "Unable to create polling threads\n" << dendl;
```

Comment:
might want to use `derr` instead of `dout(1)` for reporting an unexpected error like this.

no need to print a `\n` at the end of the log message. it's already been taken care of by the logging subsystem.